### PR TITLE
[needs-docs][processing] Optionally, also remove EMPTY geometries in RemoveNullGeometries algorithm

### PR DESCRIPTION
**Pull request by gacarrillor**
## Description

Empty geometries do not contain coordinates. Thus, like `NULL` geometries, empty geometries are not spatial data in a strict sense. To make it easier for users to clean a vector layer, the `Remove null geometries` algorithm now has an `Also remove empty geometries` option.

The algorithm dialog looks like this:

image

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmremovenullgeometry.cpp [hunk 1]
```diff
@@ -60,7 +61,10 @@ QString QgsRemoveNullGeometryAlgorithm::shortHelpString() const
 {
   return QObject::tr( "This algorithm removes any features which do not have a geometry from a vector layer. "
                       "All other features will be copied unchanged.\n\n"
-                      "Optionally, the features with null geometries can be saved to a separate output." );
+                      "Optionally, the features with null geometries can be saved to a separate output.\n\n"
+                      "If 'Also remove empty geometries' is checked, the algorithm removes features whose geometries"
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmremovenullgeometry.cpp]:
```suggestion
                      "If 'Also remove empty geometries' is checked, the algorithm removes features whose geometries "
```

File: src/analysis/processing/qgsalgorithmremovenullgeometry.cpp [hunk 2]
```diff
@@ -74,6 +78,8 @@ QVariantMap QgsRemoveNullGeometryAlgorithm::processAlgorithm( const QVariantMap
   if ( !source )
     throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
 
+  const bool remove_empty = parameterAsBoolean( parameters, QStringLiteral( "REMOVE_EMPTY" ), context );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmremovenullgeometry.cpp]:
```suggestion
  const bool removeEmpty = parameterAsBoolean( parameters, QStringLiteral( "REMOVE_EMPTY" ), context );
```

**Review by nyalldawson — COMMENTED** [review]
Nice addition!

File: src/analysis/processing/qgsalgorithmremovenullgeometry.cpp (hunk 1 above)
Comment (gacarrillor) [reply src/analysis/processing/qgsalgorithmremovenullgeometry.cpp]: Applied in https://github.com/qgis/QGIS/pull/35084/commits/3b6c04f70e57c2ce3b6bb2bee93c3f643f7a88b6

File: src/analysis/processing/qgsalgorithmremovenullgeometry.cpp (hunk 2 above)
Comment (gacarrillor) [reply src/analysis/processing/qgsalgorithmremovenullgeometry.cpp]: Applied in https://github.com/qgis/QGIS/pull/35084/commits/3b6c04f70e57c2ce3b6bb2bee93c3f643f7a88b6

**Pull request merged**